client/js/lib/statement/AssessmentTable.js

```diff
@@ -55,70 +55,73 @@ export default function AssessmentTable () {
       })
   }
 
-  window.submitForm = function (event, task) {
+  window.submitForm = function (event, task, isHashUpdateNeeded = true) {
     // In case the call originated from a native browser event it needs to be terminated
     if (event) {
       event.stopPropagation()
       event.preventDefault()
     }
 
-    const oldTarget = document.bpform.target
-    const oldAction = document.bpform.action
-    let abfrageBox
-
     const inputFields = $('form[name=bpform]').serializeArray()
     const filterOptions = inputFields.filter(inputField => inputField.name.includes('filter') && inputField.value !== '')
-
     const procedureId = $('form[name=bpform]').data('statement-admin-container')
-    window.updateFilterHash(procedureId, filterOptions)
-      .then((filterHash) => {
-        document.bpform.action = Routing.generate('dplan_assessmenttable_view_table', {
-          procedureId,
-          filterHash
-        })
-
-        let formsend = false
 
-        switch (task) {
-          case 'delete':
-            abfrageBox = dpconfirm(Translator.trans('check.entries.marked.delete'))
-            if (abfrageBox === true) {
-              document.getElementsByName('r_action')[0].value = 'delete'
-              document.bpform.submit()
-              formsend = true
-            } else {
-              return false
-            }
-            break
+    if (isHashUpdateNeeded) {
+      window.updateFilterHash(procedureId, filterOptions)
+        .then((filterHash) => {
+          document.bpform.action = Routing.generate('dplan_assessmenttable_view_table', { procedureId, filterHash })
+          handleFormSubmission(task)
+        })
+    } else {
+      handleFormSubmission(task)
+    }
+  }
 
-          case 'copy':
-            // Copying of statements within one procedure is not handled in copyStatementModal yet
-            abfrageBox = dpconfirm(Translator.trans('check.entries.marked.copy'))
-            if (abfrageBox === true) {
-              document.getElementsByName('r_action')[0].value = 'copy'
-              document.bpform.submit()
-              formsend = true
-            } else {
-              return false
-            }
-            break
+  function handleFormSubmission (task) {
```

Comment:
I really like that so much of the logic was moved into its own function which makes the handling here a lot more easy.